homeassistant/components/alert.py

```diff
@@ -0,0 +1,283 @@
+        self._skip_first = skip_first
+        self._notifiers = notifiers
+        self._can_ack = can_ack
+        self._backoff = float(backoff)
+
+        self._next_delay = None
+        self._firing = False
+        self._ack = False
+        self._cancel = None
+        self.entity_id = ENTITY_ID_FORMAT.format(slugify(entity_id))
+
+        event.async_track_state_change(hass, entity,
+                                       self.watched_entity_change)
+
+    @property
+    def name(self):
+        """Return the name of the alert."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the alert status."""
+        if self._firing:
+            if self._ack:
+                return STATE_OFF
+            return STATE_ON
+        return STATE_IDLE
+
+    @property
+    def hidden(self):
+        """Hide the alert when it is not firing."""
+        return not self._can_ack or not self._firing
+
+    @asyncio.coroutine
+    def watched_entity_change(self, entity, from_state, to_state):
+        """Determine if the alert should start or stop."""
+        _LOGGER.debug('Watched entity (%s) has changed.', entity)
+        if to_state.state == self._alert_state and not self._firing:
+            yield from self.begin_alerting()
+        if to_state.state != self._alert_state and self._firing:
+            yield from self.end_alerting()
+
+    @asyncio.coroutine
+    def begin_alerting(self):
+        """Begin the alert procedures."""
+        _LOGGER.debug('Beginning Alert: %s', self._name)
+        self._ack = False
+        self._firing = True
+        self._next_delay = self._delay
+
+        if not self._skip_first:
+            yield from self._notify()
+        else:
+            yield from self._schedule_notify()
+
+        self.hass.async_add_job(self.async_update_ha_state)
+
+    @asyncio.coroutine
+    def end_alerting(self):
+        """End the alert procedures."""
+        _LOGGER.debug('Ending Alert: %s', self._name)
+        self._cancel()
+        self._ack = False
+        self._firing = False
+        self.hass.async_add_job(self.async_update_ha_state)
+
+    @asyncio.coroutine
+    def _schedule_notify(self):
+        """Schedule a notification."""
+        self._cancel = \
+            event.async_track_time_interval(self.hass, self._notify,
+                                            self._next_delay)
+        self._next_delay = \
+            min(max(self._next_delay * self._backoff,
+                    self._delay_limits[0]),
+                self._delay_limits[1])
+
+    @asyncio.coroutine
+    def _notify(self, *args):
+        """Send the alert notification."""
+        if not self._firing:
+            return
+
+        if not self._ack:
+            _LOGGER.info('Alerting: %s', self._name)
+            for target in self._notifiers:
+                yield from self.hass.services.async_call(
+                    'notify', target, {'message': self._name})
```

Comment:
Right now the entity's friendly name is what is sent in the message. So it is both, the name and the message. Later, to implement template messages, this could be readdressed.